### ISSUE TITLE
Bug: Pagination buttons not obvious enough in list pages

### ISSUE THREAD

**Issue by asaunier** [c2corg/c2c_ui#1449]
**Describe the bug**
The current arrow buttons to go to the previous/next results page in the list pages are a bit too discrete and many users don't notice them, thinking the only results are the currently displayed ones.

**To Reproduce**
1. Go to https://www.camptocamp.org/routes?bbox=754884,5687198,774760,5717438

<img width="514" alt="Capture d’écran 2020-10-25 à 20 47 09" src="https://user-images.githubusercontent.com/1192331/97117487-5fd47800-1704-11eb-9003-14a6cdbc3d37.png">

**Expected behavior**
Making it more obvious there are other results pages.

**Comment by cbeauchesne** [comment]
I totally agree it's not easy to understand for newcomers.

Coding the solution is not hard. But finding it is not easy. Things to consider : 

* must be readable on both desktop and mobile (we can have two different solutions)
* Knowing the total count of document is important
* Vertical space is scare

And if the solution is more than a simple change in design (meaning it asks a consequent dev effort, and present a breaking change in user experience), a short presentation on forum is needed.

**Comment by ataraxido** [comment]
Hello,

What do you think to modify the behavior of this feature and to go this way :

- "1-30 sur 45701" < > becomes "45701 results" or "45701 routes"
- add an infinite scroll ( ie when the user scrolls up to the bottom of the list it automatically append 30 following results )

It could also be done in an no-auto way with an overlay "load more" button at the bottom of the list

**Comment by cbeauchesne** [comment]
The infinite scroll comes with it pitfalls. I've in mind this: 

* it will make the map not very user friendly, as icon will appears when you scroll down
* with map on mode "view only result on this map", any move of the map will reset the list to the top
* it makes hard to build an URL with an offset (yes I know, it's an edge-case usage)

I'm not saying it's not the good solution, but it's not a silver bullet. If we choose it, it'll requires a lot of attention.

> It could also be done in an no-auto way with an overlay "load more" button at the bottom of the listIt could also be done in an no-auto way with an overlay "load more" button at the bottom of the list

It'll eat some vertical space, scare ressource.

By the way, the issue is about making the button more explicit. Simply transform them in a more button-design like could do the job ?

**Comment by ataraxido** [comment]
I got your point, I didn't think of the map since I'm not using it much but you're right.

**Comment by ataraxido** [comment]
First remark : it's way more an enhancement rather than a bug

Also here my (working) proposition :

![Screenshot_2020-11-16 Sorties - Camptocamp org](https://user-images.githubusercontent.com/6763663/99266388-fdfac000-2822-11eb-8be9-42e8abd33907.png)

**Comment by cbeauchesne** [comment]
Few things to keep in mind

* having the total number of document is a requirement
* the API cannot handle an offset > 10,000 (DB liimitation)
* the API cannot deliver more than 50 documents

And I'm not really sure that getting the n-th page is really a use case (I mean, in the 80% part of the 80%/20%). Most of the time, users will need to access the next and previous page.

I think that @asaunier want to say is that it's not clear it's possible to change page, cause of the global design. but it was not about changing the ergonomy (and I'm agree with him). @asaunier do you confirm ?

**Comment by ataraxido** [comment]
After code-check and real test the API limit is 100, not 50.

Don't you think this new pagination is way more flexible and also visible/obvious ?

Working on it I tested many use cases with different displays and it seemed very useful to me, honestly.
For example displaying as a list and switching to 90 per row if you are looking for something specific.
I agree with your 80%/20% but the common next/prev is still available and much more visible, no ?
Basically it just adds possibilities without removing anything ( I'll put pack the total count ).

**Comment by cbeauchesne** [comment]
> After code-check and real test the API limit is 100, not 50.

My bad! 

 > Don't you think this new pagination is way more flexible and also visible/obvious ?

Yep, agree. I'm just always a little bit precautious with ergonomic changes.

I've added you to contributors of this repo, you can push and do a PR, it will spawn automatically a demo app, it'll be easier to have a look.

**Comment by ataraxido** [comment]
Here you go :)

**Comment by asaunier** [comment]
> I think that @asaunier want to say is that it's not clear it's possible to change page, cause of the global design. but it was not about changing the ergonomy (and I'm agree with him). @asaunier do you confirm ?

Indeed.
But I am not against changing the ergonomy if we can find a better solution.
I have added my comments to the PR at https://github.com/c2corg/c2c_ui/pull/1483#issuecomment-728243299
